certbot/certbot/_internal/main.py

```diff
@@ -1323,6 +1324,9 @@ def main(cli_args=None):
     if not cli_args:
         cli_args = sys.argv[1:]
 
+    if os.environ.get('CERTBOT_SNAPPED') == 'True':
+        cli_args = snap_config.prepare_env(cli_args)
```

Comment:
Do you think this would benefit from additional error handling, if something does go wrong in `prepare_env`? 

Say, if `snapd` is stopped or isn't ready and returns a 5xx, [this is what we get](https://gist.github.com/alexzorin/ba2e98cabf658f0463a8cafb73115d5b). IMO it would be good to only print something like:

    An error occurred while fetching Certbot snap plugins: requests.exceptions.ConnectionError: ('Connection aborted.', ConnectionRefusedError(111, 'Connection refused'))

but it does run the risk of obscuring too much without having a logging system initialized yet.

Reply:
Well, the error output that would emerge from the wrapper phase will be the first thing a regular user who just installed Certbot through snap would see. This stacktrace is hideous, I think we need to do something about it.

Indeed having a short message describing that something wrong happened, with headings, would be nice.

Given what the wrapper does, I think we can move the wrapper call after `log.pre_arg_parse_setup()` without side effects. This way, we can benefit from the minimal logging system set up at that time to handle a little better any error occurring during the initialization phase.

What do you think of this?

Reply:
I would personally prefer to be cautious; keep it as the first call and put a broad exception handler around it. But this is just paranoia, I don't see anything specifically wrong with waiting for the logging system.

Reply:
The setup is about setting some environment variables and a cli arg. None of this will impact the logging system, I think we are safe.

I move the logic after `log.pre_arg_parse_setup()`. This way we get a "nice" output and a log stored in the temporary directory when an unexpected error occurs.

Reply:
That's way better. 

1. Since we have a logger and except hook available now, shall we use it in `snap_config.py` instead of `sys.stderr.write` and `sys.exit`?

2. I think we should keep improving the below error output, so that it is clear to the user that it is `snapd`-related and not actually a network issue. Maybe if something in `requests` errors, error log an explanation (`Error occurred fetching .... make sure the snapd service is running`), then re-raise.
```
$ certbot --version
An unexpected error occurred:
requests.exceptions.ConnectionError: ('Connection aborted.', ConnectionRefusedError(111, 'Connection refused'))
Please see the logfile '/tmp/tmpiicnb1f3/log' for more details.
```

Reply:
Done.